kuma/javascript/src/header/main-menu.jsx

```diff
@@ -0,0 +1,266 @@
+                ]
+            }
+        ],
+        [locale]
+    );
+
+    /**
+     * Handles mouse and keyboard events on desktop devices
+     * @param {Object} menu - The current HTML nav element
+     */
+    function desktopInteractionHandlers(menu) {
+        menu.addEventListener('mouseover', (event: MouseEvent) => {
+            let currentTarget = event.target;
+            /* currentTarget instanceof 'HTMLButtonElement' is added
+               to keep Flow happy: https://github.com/facebook/flow/issues/218#issuecomment-74119319 */
+            if (
+                currentTarget instanceof HTMLButtonElement &&
+                currentTarget.classList.contains('top-level-entry')
+            ) {
+                if (currentTarget.nextElementSibling) {
+                    currentTarget.nextElementSibling.setAttribute(
+                        'aria-hidden',
+                        'false'
+                    );
+                }
+            }
+        });
+
+        menu.addEventListener('mouseout', (event: MouseEvent) => {
+            let currentTarget = event.target;
+            if (
+                currentTarget instanceof HTMLButtonElement &&
+                currentTarget.classList.contains('top-level-entry')
+            ) {
+                if (currentTarget.nextElementSibling) {
+                    currentTarget.nextElementSibling.setAttribute(
+                        'aria-hidden',
+                        'false'
+                    );
+                }
+            }
+        });
+    }
+
+    /**
+     * Handles touch events on mobile devices
+     * @param {Object} menu - The current HTML nav element
+     */
+    function mobileInteractionHandlers(menu) {
+        menu.addEventListener('touchend', (event: TouchEvent) => {
+            event.stopImmediatePropagation();
+            let currentTarget = event.target;
+            if (
+                currentTarget instanceof HTMLButtonElement &&
+                currentTarget.classList.contains('top-level-entry')
+            ) {
+                let subMenu = currentTarget.nextElementSibling;
+
+                if (subMenu) {
+                    let currentAriaState = subMenu.getAttribute('aria-hidden');
+                    let newAriaState =
+                        currentAriaState === 'true' ? 'false' : 'true';
+                    subMenu.setAttribute('aria-hidden', newAriaState);
+                    subMenu.classList.toggle('show');
+                }
+            }
+        });
+    }
+
+    /**
+     * Hide the visible submenu in main navigation
+     */
+    function hideSubMenus() {
+        let visibleSubMenu = document.querySelector(
+            'li ul[aria-hidden="false"]'
+        );
+        if (visibleSubMenu) {
+            visibleSubMenu.setAttribute('aria-hidden', 'true');
+            visibleSubMenu.classList.toggle('show');
+        }
+    }
+
+    useEffect(() => {
+        let menu = document.getElementById('header-main-nav');
```

Comment:
defining a ref and using the useRef().current hook is probably the more react-friendly way to do this instead of getElementById().  I'm not sure precisely what difference it makes, though.